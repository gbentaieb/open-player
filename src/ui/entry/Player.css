.ControlsWrapper {
  position: absolute;
  width: 100%;
  height: 100%;
}

.CoreWrapper {
  position: absolute;
  width: 100%;
  height: 100%;
}

.SpinnerWrapper {
  position: absolute;
  width: 100%;
  height: 100%;
  display: flex;
  align-items: center;
  justify-content: center;
}
