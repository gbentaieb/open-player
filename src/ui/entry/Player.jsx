import React, { Component } from 'react';
import PropTypes from 'prop-types';
import MuiThemeProvider from '@material-ui/core/styles/MuiThemeProvider';
import { createMuiTheme } from '@material-ui/core/styles';

import { setConfig } from '../actions/ConfigActions';
import { requestPlay } from '../actions/CoreActions';

import CorePlayerContainer from '../components/container/coreplayer/CorePlayerContainer';
import ControlsContainer from '../components/container/controlscontainer/ControlsContainer';

import style from './Player.css';
import SpinnerContainer from '../components/container/spinner/SpinnerContainer';

const defaultTheme = createMuiTheme({
  typography: {
    useNextVariants: true,
  },
});

class Player extends Component {
  static propTypes = {
    videoElement: PropTypes.object.isRequired,
    config: PropTypes.object.isRequired,
    onLoad: PropTypes.func.isRequired,
  };

  static contextTypes = {
    store: PropTypes.object,
  };

  state = {
    theme: createMuiTheme(defaultTheme),
  }

  componentWillMount() {
    const { config } = this.props;

    if (config.mainColor) {
      this.setMainColor(config.mainColor);
    }
  }

  componentDidMount() {
    const { config } = this.props;
    this.context.store.dispatch(setConfig(config));
    this.props.onLoad(this);
  }

  componentWillReceiveProps(nextProps) {
    const { config } = this.props;

    if (nextProps.config !== config) {
      this.context.store.dispatch(setConfig(nextProps.config));
    }
  }

  setMainColor(color) {
    const palette = {
      primary: {
        main: color,
      },
    };

    this.setState(state => ({
      ...state,
      theme: createMuiTheme({ ...defaultTheme, palette }),
    }));
  }

  requestPlay(play) {
    this.context.store.dispatch(requestPlay(play));
  }

  render() {
    return (
      <MuiThemeProvider theme={this.state.theme}>
        <div>
          <div className={style.ControlsWrapper}>
            <ControlsContainer />
          </div>
          <div className={style.CoreWrapper}>
            <CorePlayerContainer videoElement={this.props.videoElement} />
          </div>
          <div className={style.SpinnerWrapper}>
            <SpinnerContainer />
          </div>
        </div>
      </MuiThemeProvider>
    );
  }
}

export default Player;
