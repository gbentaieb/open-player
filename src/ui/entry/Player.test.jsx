import React from 'react';
import { Provider } from 'react-redux';
import Player from './Player';

const createFakeStore = state => (
  {
    default: () => {},
    subscribe: () => {},
    dispatch: () => {},
    getState: () => (
      { ...state }
    ),
  }
);

const defaultStore = createFakeStore({
  config: {},
});

describe('test Player component', () => {
  test('Player > Snapshot', () => {
    const globalWrapper = shallow(
      <Provider store={defaultStore}>
        <Player videoElement={{}} config={{}} onLoad={() => {}} />
      </Provider>,
    );

    const wrapper = globalWrapper.find(Player);
    expect(wrapper).toMatchSnapshot();
  });
});
