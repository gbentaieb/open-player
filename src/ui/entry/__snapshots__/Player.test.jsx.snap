// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Player component Player > Snapshot 1`] = `
<Player
  config={Object {}}
  onLoad={[Function]}
/>
`;
